.builder/actions/aws_crt_python.py

```diff
@@ -1,248 +1,72 @@
-
-        # note: not using `ldconfig --print-cache` to find it because
-        # some installers put it in weird places where ldconfig doesn't look
-        # (like in a subfolder under lib/)
-
-        for lib_dir in ['lib64', 'lib']:  # search lib64 before lib
-            for base_dir in ['/usr/local', '/usr', '/', ]:
-                search_dir = os.path.join(base_dir, lib_dir)
-                for root, dirs, files in os.walk(search_dir):
-                    for file_name in files:
-                        if 'libsofthsm2.so' in file_name:
-                            return os.path.join(root, file_name)
-        return None
-
-    def _exec_softhsm2_util(self, *args, **kwargs):
-        if 'check' not in kwargs:
-            kwargs['check'] = True
-
-        result = self.env.shell.exec('softhsm2-util', *args, **kwargs)
-
-        # older versions of softhsm2-util (2.1.0 is a known offender)
-        # return error code 0 and print the help if invalid args are passed.
-        # This should be an error.
-        #
-        # invalid args can happen because newer versions of softhsm2-util
-        # support more args than older versions, so what works on your
-        # machine might not work on some ancient docker image.
-        if 'Usage: softhsm2-util' in result.output:
-            raise Exception('softhsm2-util failed')
-
-        return result
-
-    def _find_sofhsm2_token_slot(self):
-        """Return slot ID of first initialized token"""
-
-        output = self._exec_softhsm2_util('--show-slots').output
-
-        # --- output looks like ---
-        # Available slots:
-        # Slot 0
-        #    Slot info:
-        #        ...
-        #        Token present:    yes
-        #    Token info:
-        #        ...
-        #        Initialized:      yes
-        current_slot = None
-        current_info_block = None
-        for line in output.splitlines():
-            # check for start of "Slot <ID>" block
-            m = re.match(r"Slot ([0-9]+)", line)
-            if m:
-                current_slot = m.group(1)
-                current_info_block = None
-                continue
-
-            if current_slot is None:
-                continue
-
-            # check for start of next indented block, like "Token info"
-            m = re.match(r"    ([^ ].*)", line)
-            if m:
-                current_info_block = m.group(1)
-                continue
-
-            if current_info_block is None:
-                continue
-
-            # if we're in token block, check for "Initialized: yes"
-            if "Token info" in current_info_block:
-                if re.match(r" *Initialized: *yes", line):
-                    return current_slot
-
-        raise Exception('No initialized tokens found')
-
-
-class AWSCrtPython(Builder.Action):
+            print("Could not update pip via normal pip upgrade. Next trying via package manager...")
```

Comment:
whoa, that's not the line I meant to highlight. I meant to say, call `shell.exec(..., check=False)`  instead of doing `Builder.Script(commands=[[...]], exit_on_fail=False).run(env)`